indexes/utxoindex/src/update_container.rs

```diff
@@ -0,0 +1,113 @@
+            supply_change: 0,
+            tips: BlockHashSet::new(),
+        }
+    }
+
+    /// Add a [`UtxoDiff`] the the [`UtxoIndexChanges`] struct.
+    pub fn add_utxo_diff(&mut self, utxo_diff: UtxoDiff) {
+        let (to_add, mut to_remove) = (utxo_diff.add, utxo_diff.remove);
+
+        for (transaction_outpoint, utxo_entry) in to_add.into_iter() {
+            if to_remove.remove(&transaction_outpoint).is_some() {
+                continue;
+            }; // We try and remove from `utxo_diff.remove`, if we do, discard utxo.
+
+            self.supply_change += utxo_entry.amount as CirculatingSupplyDiff; // TODO: Using `virtual_state.mergeset_rewards` might be a better way to extract this.
+
+            match self.utxo_changes.added.entry(utxo_entry.script_public_key) {
+                Entry::Occupied(mut entry) => {
+                    entry.get_mut().insert(
+                        transaction_outpoint,
+                        CompactUtxoEntry::new(utxo_entry.amount, utxo_entry.block_daa_score, utxo_entry.is_coinbase),
+                    );
+                }
+                Entry::Vacant(entry) => {
+                    let mut value = CompactUtxoCollection::with_capacity(1);
+                    value.insert(
+                        transaction_outpoint,
+                        CompactUtxoEntry::new(utxo_entry.amount, utxo_entry.block_daa_score, utxo_entry.is_coinbase),
+                    );
+                    entry.insert(value);
+                }
+            };
+        }
+
+        for (transaction_outpoint, utxo_entry) in to_remove.into_iter() {
+            self.supply_change -= utxo_entry.amount as CirculatingSupplyDiff; // TODO: Using `virtual_state.mergeset_rewards` might be a better way to extract this.
+
+            match self.utxo_changes.removed.entry(utxo_entry.script_public_key) {
+                Entry::Occupied(mut entry) => {
+                    entry.get_mut().insert(
+                        transaction_outpoint,
+                        CompactUtxoEntry::new(utxo_entry.amount, utxo_entry.block_daa_score, utxo_entry.is_coinbase),
+                    );
+                }
+                Entry::Vacant(entry) => {
+                    let mut value = CompactUtxoCollection::with_capacity(1);
+                    value.insert(
+                        transaction_outpoint,
+                        CompactUtxoEntry::new(utxo_entry.amount, utxo_entry.block_daa_score, utxo_entry.is_coinbase),
+                    );
+                    entry.insert(value);
+                }
+            };
+        }
+    }
+
+    /// Add a [`Vec<(TransactionOutpoint, UtxoEntry)>`] the the [`UtxoIndexChanges`] struct
+    ///
+    /// Note: This is meant to be used when resyncing.
+    pub fn add_utxo_collection_vector(&mut self, utxo_vector: Vec<(TransactionOutpoint, UtxoEntry)>) {
+        let mut circulating_supply: CirculatingSupply = 0;
```

Comment:
Don't fully understand this, but for me it seems like this is asking to merge both `add_utxo_diff`, and `update_utxo_diff` into one method? but I require two methods, as during resync I pass a `Vec<(TransactionOutpoint, UtxoEntry)>` instead of a `UtxoCollection`, I could use one method that takes an iterator as an argument, but then I wouldn't be able to remove utxos from `to_remove` in `update_utxo_diff` by utilizing the hashmap.